api/services/gameService.js

```diff
@@ -106,62 +79,39 @@ module.exports = {
-                  message: "Can't populate game, because it does not have players collection",
-                });
-              })
-              // then find points
-              .then(function findPoints(values) {
-                const [game, p0, p1, spectators] = values;
-                const p0Points = cardService.findPoints({ userId: p0.id });
-                const p1Points = cardService.findPoints({ userId: p1.id });
-                return Promise.all([
-                  Promise.resolve(game),
-                  Promise.resolve(p0),
-                  Promise.resolve(p1),
-                  Promise.resolve(spectators),
-                  p0Points,
-                  p1Points,
-                ]);
-              })
-              // then format results & resolve
-              .then(function finish(values) {
-                const [game, p0, p1, spectators, p0Points, p1Points] = values;
-                const populatedP0 = new tempUser(p0, p0Points);
-                const populatedP1 = new tempUser(p1, p1Points);
-                const result = new tempGame(game, spectators, populatedP0, populatedP1);
-                return resolve(result);
-              })
-              .catch(function failed(err) {
-                reject(err);
-              })
-          );
+   populateGame: async function(options) {
+    try{
+      if (!options) {
+        if (!Object.hasOwnProperty.call(options, 'gameId') && !typeof options.gameId === 'number') {
+          throw new Error({message: 'gameId is required and must be a number'});
         }
-        reject({ message: 'gameId is required and must be a number' });
-      } else {
-        reject({ message: 'Cannot populate Game without GameId (options had no gameId)' });
+        throw new Error({ message: 'Cannot populate Game without GameId (options had no gameId)' }); 
+      } 
+      // find game
+      const game = await gameService.findGame({ gameId: options.gameId });
+      
+      if (!game.players) {
+        if (game.players.length < 2) {
+          throw new Error({message: 'Cannot populate game without two players'});
+        }
+        throw new Error({message: 'Cannot populate game, because it does not have players collection'});
       }
-    });
-  }, //End populateGame()
+      //find users and points
+      const [p0, p1, spectatingUsers, p0Points, p1Points] = await Promise.all([
+        userService.findUser({ userId: game.players[0].id }),
+        userService.findUser({ userId: game.players[1].id }),
+        fetchSpectatorUsernames(game.id),
+        cardService.findPoints({ userId: game.players[0].id }),
+        cardService.findPoints({ userId: game.players[1].id })
+      ]);
+      // then format results
+      const populatedP0 = formatPlayerData(p0, p0Points);
+      const populatedP1 = formatPlayerData(p1, p1Points);
+      return {...game, players: [populatedP0, populatedP1], spectatingUsers};
+    } catch(err){ 
+      return err.message;
+     }
```

Comment:
So the current implementation errors (with `reject()`) if there is an error and leaves it up to the caller to sort it out. We could keep that the same here by just omitting the try/catch block altogether. I think it's reasonable in this case to let the caller's manage the error handling. We can add it back with some proper error logging after completing #413